Makefile

```diff
@@ -49,6 +49,7 @@ clean:
 	coverage erase
 
 test: clean
+	curl --silent --head http://localhost:9223/roster_test > /dev/null || make test.run_elasticsearch  # Launch ES if needed
```

Comment:
This is really helpful locally, but fails in Travis; if someone knows how to make it work there, that'd be great, otherwise I'll pull it before merging

Reply:
it looks like the error in travis is saying there's a directory missing, a directory that is created locally by running the `make test.install_elasticseach` command (https://github.com/edx/edx-analytics-data-api/blob/9190ca68e2bf9033b019dfd9a6f7c7ba84d4ff0a/Makefile#L19)

i wonder if it makes sense to have `test.run_elasticsearch` run the `test.install_elasticsearch` logic if the folder is not found?

that said though, because we haven't hit this problem in the past, running ES in travis may not be necessary (specifically if we were not running it before), and would just be adding this to travis only to make local dev a little more convenient

what do you think?

Reply:
If there's an easy way to command to check for "am I on Travis?" I could probably add that in to avoid the launch attempt; do you know of something I could check for?

Reply:
Maybe just "if that directory doesn't exist, don't try" (since it would fail locally too)

Reply:
I haven't been following closely, but Travis has environment variables it includes that you can check for: https://docs.travis-ci.com/user/environment-variables#default-environment-variables

Reply:
I added a check for the local ES install and only attempt to launch from there if it exists.  That passes Travis.

Reply:
Cool to see you got that working! Nice!